lib/Common/Memory/WriteBarrierMacros.h

```diff
@@ -17,6 +17,8 @@
 
 #define FieldWithBarrier(type, ...) \
     typename WriteBarrierFieldTypeTraits<type, ##__VA_ARGS__>::Type
+#define FieldWithBarrierT(type, ...) \
```

Comment:
only need to remove the typename for legacy compiler right? if so can you use _MSC_VER to differentiate it so you don't need to change FunctionBody.h?